Add native token how-tos (#541)
iotaledger/iota-sdk: documentation/sdk/docs/how_tos/native_tokens/burn.mdx

---
title: Burn Native Tokens
sidebar_label: Burn
description: 'How to burn native tokens'
image: /img/logo/iota_mark_light.png
keywords:
- how to
- burn
- burn native token
- foundry
- nodejs
- python
- rust
---

import CodeBlock from '@theme/CodeBlock';
import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";
import NodejsCode from '!!raw-loader!../../../../../bindings/nodejs/examples/how_tos/native_tokens/burn.ts';
import PythonCode from '!!raw-loader!../../../../../bindings/python/examples/how_tos/native_tokens/burn.py';
import RustCode from '!!raw-loader!../../../../../sdk/examples/how_tos/native_tokens/burn.rs';
import DestroyFoundry from '../../_admonitions/_destroy_foundry.md';

You may want to burn some native tokens. To do so, you will need to call the
`Account.burn(burn, options)` function.

<DestroyFoundry/>

## Code Example

The following example will:

1. Create a wallet.
2. Get Alice's account which was [created in the first guide](../accounts_and_addresses/create_account.mdx).
3. [Get the account's balance](../accounts_and_addresses/check_balance.mdx).

> 4. Burn 1 native tokens of the first ID with enough funds.

Reviewer: ```suggestion
4. Burn 1 native token of the first ID with enough funds.
```

---
title: Burn Native Tokens
sidebar_label: Burn
description: 'How to burn native tokens'
image: /img/logo/iota_mark_light.png
keywords:
- how to
- burn
- burn native token
- foundry
- nodejs
- python
- rust
---

import CodeBlock from '@theme/CodeBlock';
import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";
import NodejsCode from '!!raw-loader!../../../../../bindings/nodejs/examples/how_tos/native_tokens/burn.ts';
import PythonCode from '!!raw-loader!../../../../../bindings/python/examples/how_tos/native_tokens/burn.py';
import RustCode from '!!raw-loader!../../../../../sdk/examples/how_tos/native_tokens/burn.rs';

> You may want to burn some of your circulating native tokens. To do so, you will need to call the

Reviewer: ```suggestion
You may want to burn some native tokens. To do so, you will need to call the
```

`Account.burn(burn, options)` function.

:::warning

> If you burn your native tokens, you will not be able to destroy the foundry that created them. If you want to destroy

Reviewer: Since the native tokens could also be minted by someone else, it should maybe be something like this
```suggestion
If you burn native tokens, it will be impossible to destroy the foundry that created them and get its storage deposit amount back. If you control the foundry and want to destroy
```

Author: Sounds good

the foundry in the future, you should consider [melting your native tokens](decrease_supply.mdx)

:::

## Code Example

The following example will:

> 1. Create an account manager.

Reviewer: No

Author: I will never learn it :sadness:
```suggestion
1. Create a wallet.
```
